src/blob.rs

```diff
@@ -391,7 +394,9 @@ impl<'a> BlobObject<'a> {
                 MediaQuality::Worse => WORSE_AVATAR_SIZE,
             };
 
-        self.recode_to_size(context, blob_abs, img_wh).await
+        // max_bytes is 20_000 bytes as that's the max header size of some servers
```

Comment:
The real explanation is: outlook servers don't allow headers larger than 32k. 32 / 4 * 3 = 24k if you account for base64 encoding. To be safe, we reduced this further to 20k.